hashintel/hash
# H-1156: Update browser extension to use new Quick Note type

## 🌟 What is the purpose of this PR?




This PR updates the browser extension to create instances of the `Quick Note` system type in HASH.

## 🔗 Related links





- H-1156

## 🔍 What does this change?




- moves GQL queries into dedicated files
- adds codegen so that the GQL query/mutation variables and return types can be typed using TS
- adds the following local dependencies:
  - `@local/hash-subgraph`
  - `@local/hash-isomorphic-utils`
  - `@blockprotocol/type-system`
- upgrades `@graphql-codegen/cli` (added this package via `yarn add`, and thought it might be worth upgrading as we are 3 releases behind the latest one)

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ⚠️ Known issues



## 🐾 Next steps



## 🛡 What tests cover this?



None

## ❓ How to test this?



1. Run the browser plugin locally
2. Create a quick note (you can test out multiple paragraphs)
3. Check in the HASH instance used by the browser plugin whether the quick note has been created

## 📹 Demo



https://github.com/hashintel/hash/assets/42802102/9f9d78e7-3715-4681-8ceb-b67385a2d07b
